Packs/DemistoLocking/Integrations/DemistoLock/DemistoLock.js

```diff
@@ -57,33 +56,63 @@ switch (command) {
             logDebug('Task guid: ' + guid + ', Incident:' + incidentID + ' | Current lock is: ' + JSON.stringify(lock) + ', version: ' + version);
 
             if (lock.guid === guid) {
-                break;
+                var md = '### Demisto Locking Mechanism\n';
+                md += 'Lock acquired successfully\n';
+                md += 'GUID: ' + guid;
+                return { ContentsFormat: formats.markdown, Type: entryTypes.note, Contents: md };
             }
-            if (!lock.guid) {
-                try {
-                    setLock(guid, lockInfo, version);
-                } catch(err) {
-                    logDebug('Task guid: ' + guid + ', Incident:' + incidentID + ' | Failed setting lock: ' + err.message);
+            else {
+                // attempt to acquire the lock
+                if (!lock.guid) {
+                    try {
+                        setLock(guid, lockInfo, version);
+                    } catch (err) {
+                        logDebug(err.message)
+                    }
+                }
+                var timeout_err_msg = 'Timeout waiting for lock\n';
+                timeout_err_msg += 'Lock name: ' + lockName + '\n';
+                timeout_err_msg += 'Lock info: ' + lock.info + '\n';
+                return {
+                    Type: entryTypes.note,
+                    Contents: 'Lock was not acquired, Polling.',
+                    PollingCommand: 'demisto-lock-get',
+                    NextRun: '30',
+                    PollingArgs: { name: lockName, info: args.info, timeout: args.timeout, guid: guid, timeout_err_msg: timeout_err_msg },
```

Comment:
Sending the custom message won't break polling?

Reply:
No, it works as expected. The `timeout_err_msg` argument is a preparation for overriding the default timeout message.